Finish user authentication

DaveZag opened the pull request:
# Finish user authentication
### On this branch I:
- Add user authentication Endpoints.
- Set up `devise-jwt` for web tokens.
- Overwrite `Devise` registration and session controllers.
- Add `jti` to the users' table to prevent replay attacks by preventing the same JWT from being replayed.
- Remove the food_orders table and all its associations.
- Run migrations to set up the updated associations between User, Food, and Order.

![1](https://user-images.githubusercontent.com/107707705/225657585-51b718ef-1186-4186-8faf-0ef33c1f8451.PNG)

DaveZag committed:
Overwrite devise User sessions and registrations controllers to use our own.

DaveZag committed:
add devise jwt configurations

DaveZag committed:
Add serializer fot the User model

Fix the session error when signup a user

Add methods in controllers to handle signup, login an logout

DaveZag committed:
Fix linters errors

Kaweesi-Matia reviewed (approved):
Hi @DaveZag 
Your implementation for the user authentication of the APP is complete! There is nothing else to say other than... it's time to merge it :shipit:

Congratulations! 🎉
<img src="https://camo.githubusercontent.com/bf83cbc7d47fad6100993d4560eeb6269a3e575766f5cc0043c5d4f18bf1e919/68747470733a2f2f6369352e676f6f676c6575736572636f6e74656e742e636f6d2f70726f78792f7957676a5273523176705538336b4477733239694e504c30306c4c2d4232386c4171735a744d4b303134504d3279784a524e624c5f7a72396e3176396734536d657455624a31774f7a6434374571304334647568527732555a524368547331396236707a7073584b687849656d5f78634e527252436d636b7261764e6e5578454b7a4736787a43783731356b78453042686a68644776674c30325a4972474c6c434a68682d6c4f4476796a73443049504a63647443304f4261315a58596f75697965505a7930393071484d79492d34494654455f726e746a635461334d57516e594c6148444a4d4a6f68456163477546303363437973332d336559424a587a364e69383968554979524c6f73396e714d325a32454472635739666348784b69726d33334c37706256673634774376356c5f6a4c78675f5a4279616e3044413d73302d642d65312d66742368747470733a2f2f63616d6f2e67697468756275736572636f6e74656e742e636f6d2f653163363032396366343065376634333336316331346364373639316462366665626666663531633234633030616436373230663937376631323430623135392f36383734373437303733336132663266366436353634363936313330326536373639373036383739326536333666366432663664363536343639363132663634353935613735373134613463343435363733353734643463353737393439373834613266363736393730363837393265363736393636" />
### Highlights

- All linters are passing. ✔️

- [ ] Nice work on implementing all features of the user authentication. it looks amazing. ✔️

Cheers and happy coding 👏 👏 👏,
Kaweesi Matia

DaveZag committed:
Remove food_orders table and associations

Configure new associations between Users, Foods and Orders and run migrations

Fix linters errors
